### PR TITLE
feat: allow controllers to call `__motoko_stable_var_info` query endpoint

### PR DESCRIPTION
**Pull request by ggreif**
This was originally spec-ed, but there was no possibility to implement it (a canister couldn't get its own controller list).
In the meantime canisters can get their controller list (via query), but a query cannot call another query 🫤.

So we had to wait until the `ic0.is_controller` syscall became available. Now it is there and the implementation is easy.

(It was technically possible to call it before, but it invariably trapped.)

### CODE REVIEW

File: src/lowering/desugar.ml [hunk 1]
```diff
@@ -438,8 +438,9 @@ and export_footprint self_id expr =
   ([ letD (var v typ) (
        funcE v (Shared Query) Promises [bind1] [] [ret_typ] (
            (asyncE T.Fut bind2
-              (blockE [expD (assertE (primE (I.RelPrim (caller, Operator.EqOp))
-                                        [primE I.ICCallerPrim []; selfRefE caller]));
+              (blockE [expD (assertE (orE (primE (I.RelPrim (caller, Operator.EqOp))
```
Comment (crusso) [inline src/lowering/desugar.ml:441]: Might be cheaper to call ICCallerPrim just once, but don't think it matters.

**Comment by crusso** [comment]
I guess we could now provide a helper in ExperimentalInternetComputer to call this on another canister than self, but not urgent.

File: src/lowering/desugar.ml (hunk 1 above)
Comment (ggreif) [reply src/lowering/desugar.ml:441]: I am not overly concerned :-)

Comment (ggreif) [reply src/lowering/desugar.ml:441]: This is now #4108!

**Pull request merged**